salt/modules/x509.py

```diff
@@ -564,12 +549,20 @@ def get_public_key(key):
 
         salt '*' x509.get_public_key /etc/pki/mycert.cer
     '''
-    text = _text_or_file(key)
 
-    text = get_pem_entry(text)
+    if isinstance(key, M2Crypto.X509.X509):
+        rsa = key.get_pubkey().get_rsa()
+        text = "None"
```

Comment:
An empty string is more correct, and I've changed it. It needs to be a string so that the startswith comparisons can be done.